dvc/repo/stage.py

```diff
@@ -473,6 +473,9 @@ def is_out_or_ignored(root, directory):
         for root, dirs, files in self.repo.dvcignore.walk(
             self.fs, self.repo.root_dir
         ):
+            logger.trace(  # type: ignore[attr-defined]
+                "Collect repo walking '%s'", root
+            )
```

Comment:
Hey, @Erotemic, What do you think of introducing a logger that also tracks the duration of the walk instead?

```python
walk_iter = self.repo.dvcignore.walk(self.fs, self.repo.root_dir)
if logger.isEnabledFor(logging.TRACE):
    walk_iter = log_walk(walk_iter)

for root, dirs, files in walk_iter:
    ...
```

Let me know, otherwise, we can take over. Thanks. Sorry for the late review.

Reply:
Also, we could repurpose some code from [funcy.log_iter_durations](https://github.com/Suor/funcy/blob/60910f8e601c9112312a02037a9c7d9c465e2daf/funcy/debug.py#L186), especially the `format_time`.

Reply:
I'm not sure how relevant logging the time is, unless DVC could use that to improve the overall speed of this step (it's getting quite sluggish as my DVC repo size increases). The motivating use case here is to show a user where DVC is spending a lot of time --- its quite obvious what is taking time without explicit logging of it because you see the same root directory over and over. Also, the time for each iteration will be roughly the same, so it wont help cue a user into what directory is causing the issue.

Overall, I'm not opposed to logging the time, I would just rather see any indication of what directory is causing the issue in the new release sooner rather than later.

On the other hand, if DVC used that time information to *notice* that a directory was taking a long time and then printed out an INFO statement to inform the user without `-vv` being given, then I think that would be useful, but maybe better suited for another PR. I don't have time to implement that myself.

As an aside, the logging in funcy probably should be using `time.perf_counter` (or `time.perf_counter_ns` if available) instead of `time.time`.